fontdrasil/src/orchestration.rs

```diff
@@ -0,0 +1,102 @@
+//! Common parts of work orchestration.
+
+use std::{
+    collections::HashSet,
+    fmt::Debug,
+    hash::Hash,
+    sync::{Arc, RwLock},
+};
+
+pub const MISSING_DATA: &str = "Missing data, dependency management failed us?";
+
+/// A unit of work safe to run in parallel
+///
+/// Naively you'd think we'd just return FnOnce + Send but that didn't want to compile
+/// See <https://github.com/rust-lang/rust/issues/29625>.
+///
+/// Data produced by work is written into a Context (type parameter C).
+pub trait Work<C, E> {
+    fn exec(&self, context: &C) -> Result<(), E>;
+}
+
+#[derive(Clone)]
+pub struct Cache<T: Default> {
```

Comment:
Unless absolutely required, preference is to avoid trait bounds on type declarations, since they're sort of 'contagious'. The preferred approach is to keep the trait bounds to the impl blocks, where you can pick exactly the set of constraints that are necessary for that impl.

I presume that you had some annoyance around `Default`, here?

In any case this is a funny type. Is `T` supposed to be some sort of collection?

Reply:
In looking again I think I originally had a slightly more interesting idea but ended up with what amounts to a verbose type alias, will delete.